api/src/pcapi/core/users/api.py

```diff
@@ -1147,3 +1148,36 @@ def save_flags(user: models.User, flags: dict) -> None:
                 save_firebase_flags(user, value)
             case _:
                 raise ValueError()
+
+
+class DeviceId(pydantic.BaseModel):
+    if typing.TYPE_CHECKING:  # https://github.com/pydantic/pydantic/issues/156
+        device_id: str
+    else:
+        device_id: pydantic.constr(min_length=1, strip_whitespace=True)
+
+
+def save_trusted_device(device_info: "account_serialization.TrustedDevice | None", user: models.User) -> None:
+    if device_info is None:
+        return
+
+    try:
+        DeviceId(device_id=device_info.device_id)
+    except pydantic.error_wrappers.ValidationError:
```

Comment:
Ça m'a l'air un poil overkill de définir un validateur pydantic pour vérifier que le "device_id" n'est pas vide (ou composé seulement d'espaces). Dans un commentaire plus bas, j'ai indiqué comment `account_serialization.TrustedDevice` pourrait automatiquement supprimer les espaces. Ainsi, on pourrait ici faire simplement : 

```suggestion
    if not device_info.device_id:
        logger.info("Invalid deviceId was provided...", ...)
        return
```